docs/discover/index.rst

```diff
@@ -0,0 +1,213 @@
+longer than that to produce their response; but given the simple
+constraints we support so far, we would probably consider them broken in
+that case.  Reducing the timeout to just a few seconds will make pyVO
+continue earlier on broken services.  But that of course increases the
+risk of cutting off working services.
+
+If in doubt, have a brief look at the log lines; if a service that
+sounds promising shows a timeout, perhaps try again with a longer
+timeout or use partial matching.
+
+
+Overriding service selection
+----------------------------
+
+You can also pass a `pyvo.registry.RegistryResults` instance to
+``services`` to override the automatic selection of services to query.
+See the discussion of overriding the service selection in Discoverers_.
+
+
+Discoverers
+===========
+
+For finer control of the discovery process, you can directly use
+the `pyvo.discover.image.ImageDiscoverer` class.  It is constructed with
+essentially the same parameters as the search function.
+
+To run the discovery, first establish which services to query.  There
+are two ways to do that:
+
+* Call the ``discover_services()`` method.  This is what the search
+  function does; it uses your constraints as above.
+* Pass a `pyvo.registry.RegistryResults` instance to ``set_services``.
+  This lets you do your own searches.  The image discoverer will only
+  use resources that it knows how to handle.  For instance, it is safe
+  to call something like::
+
+    discoverer.set_services(
+      registry.search(registry.Author("Hubble, %")))
+
+  to query services that give a particular author.  More realistically,
+
+  ::
+
+    discoverer.set_services(
+      registry.search(registry.Datamodel("obscore")))
+
+  will restrict the operation to obscore services.
+
+``set_services`` will purge redundant services, which means that
+services that say they (or their data) is served by another service that
+will already be queried will not be queried.  Outside of debugging, this
+is what you want, but if you really do not want this, you can pass
+``purge_redundant=False``.  Note, however, that you will still get only
+one match per access URL of the dataset.
+
+Once you have set the services, call ``query_services()`` to fill the
+``results`` and ``log_messages`` attributes.  It may be informative to
+watch these change from, say, a different thread.  Changing their
+content has undefined results.
+
+A working example would look like this::
+
+  from pyvo import discover, registry
+  from astropy.time import Time
+
+  im_discoverer = discover.image.ImageDiscoverer(
+    space=(274.6880, -13.7920, 0.1),
+    time=(Time('1996-10-04'), Time('1996-10-10')))
+  im_discoverer.set_services(
+    registry.search(keywords=["heasarc rass"]))
+  im_discoverer.query_services()
+  print(im_discoverer.log_messages)
+  print(im_discoverer.results)
+
+
+
+Reference/API
+=============
+
+.. automodapi:: pyvo.discover.image
```

Comment:
image is not a public namespace, expose only the public API from the module into the user docs
```suggestion
.. automodapi:: pyvo.discover
```

Reply:
Ok, thanks.